### PR TITLE
Paste image to card description

### PR DESCRIPTION
**Pull request by RijaRR**
* Resolves: https://github.com/nextcloud/deck/issues/533

### Summary
on paste image in edition of card description. We insert image as card attachement and in card description. PR related to this issue 

### Checklist

- [ ] Code is properly formatted
- [ ] Sign-off message is added to all commits
- [ ] Tests (unit, integration, api and/or acceptance) are included
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: src/components/card/Description.vue [hunk 1]
```diff
@@ -212,18 +224,35 @@ export default {
...
 				embed = '!'
 			}
 			const attachmentString = embed + '[📎 ' + attachment.data + '](' + this.attachmentPreview(attachment) + ')'
-			const newContent = descString + '\n' + attachmentString
+			const cursor = this.$refs.markdownEditor.easymde.codemirror.getCursor()
+			const descStringLines = this.$refs.markdownEditor.easymde.value().split('\n')
+			let lineUpatedLength = 0
+
+			for (let i = 0; i < descStringLines.length; i++) {
+				if (i === cursor.line) {
```
Comment (marcelklehr) [inline src/components/card/Description.vue:238]: Using a for-loop here seems excessive, if you only want to change one line. How anout this?

```js
if (descStringLines[cursor.line]) {
        const line = descStringLines[cursor.line]
	descStringLines[cursor.line] = line.substring(0, cursor.ch) + attachmentString + line.substring(cursor.ch)
	lineUpatedLength = descStringLines[cursor.line].length
}
```

File: src/components/card/Description.vue [hunk 2]
```diff
@@ -263,6 +292,40 @@ export default {
...
+			}
+
+			if (files.length === 0) {
+				return
+			}
+
+			this.loading = true
+			event.preventDefault()
+			for (const file of files) {
+				this.onLocalAttachmentSelected(file, 'file').then(this.addLastAttachmmentOnCursor)
```
Comment (marcelklehr) [inline src/components/card/Description.vue:326]: Phew, this method should probably have been named differently, heh :D

File: src/components/card/AttachmentUploadProgress.vue [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+				<progress :value="attachment.progress" max="100" />
+			</a>
+		</div>
+	</div>
+</template>
+
+<script>
+
+export default {
+	name: 'AttachmentUploadProgress',
```
Comment (marcelklehr) [inline src/components/card/AttachmentUploadProgress.vue:40]: Good thinking to extract this component! :+1:

File: src/components/card/Description.vue (hunk 1 above)
Comment (RijaRR) [reply src/components/card/Description.vue]: @marcelklehr it should work. I've changed this way.

File: src/components/card/AttachmentUploadProgress.vue [hunk 4]
```diff
@@ -0,0 +1,55 @@
+<!--
+  - @copyright Copyright (c) 2022 Julius Härtl <rija.propitech@gmail.com>
+  -
+  - @author Julius Härtl <rija.propitech@gmail.com>
```
[the remark this answers is not in the document]
Comment (RijaRR) [reply src/components/card/AttachmentUploadProgress.vue:4]: @marcelklehr I'm not sure which one is right. The content is coming from a previous file though AttachmentUploadProgress.vue will be created in this PR.

[the remark this answers is not in the document]
Comment (marcelklehr) [reply src/components/card/AttachmentUploadProgress.vue:4]: Thanks!

File: src/components/card/Description.vue [hunk 5]
```diff
@@ -263,6 +290,40 @@ export default {
...
+				return
+			}
+			this.addAttachment(this.$store.getters.attachmentsByCard(this.card.id)[attachementLength - 1])
+		},
+		onPaste() {
+			if (!this.descriptionEditing) {
+				return
+			}
+
+			const clipboardData = event.clipboardData
```
Comment (marcelklehr) [inline src/components/card/Description.vue:305]: Where does the variable `event` come from?

File: src/components/card/Description.vue (hunk 2 above)
Comment (marcelklehr) [reply src/components/card/Description.vue:324]: Not your fault though, of course

File: src/components/card/Description.vue [hunk 6]
```diff
@@ -263,6 +290,40 @@ export default {
...
+					}
+				}
+			}
+
+			if (files.length === 0) {
+				return
+			}
+
+			this.loading = true
+			event.preventDefault()
```
Comment (marcelklehr) [inline src/components/card/Description.vue:322]: This doesn't seem to work in Firefox as I still receive a path to my file in the text of the document:

image

**Comment by marcelklehr** [comment]
In chrome nothing is inserted at all for me, for some reason :/

**Comment by okow** [comment]
Any progress here?
I would *love* to finally see this feature be implemented...

**Comment by RijaRR** [comment]
Thank you for considering my contribution. After further consideration, I have decided to withdraw my pull request. I believe that other developers can better address the changes or additions needed for this project. I look forward to seeing how the project progresses. Feel free to continue the discussion without me. Best wishes to the team!

**Pull request closed without merging**